thrust/cmake/ThrustHeaderTesting.cmake

```diff
@@ -146,6 +146,19 @@ foreach(thrust_target IN LISTS THRUST_TARGETS)
     "CUB_WRAPPED_NAMESPACE=wrapped_cub")
   thrust_add_header_test(${thrust_target} base "${header_definitions}")
 
+  # We need to ensure that the different dispatch mechanisms work
+  set(header_definitions
+    "THRUST_WRAPPED_NAMESPACE=wrapped_thrust"
+    "CUB_WRAPPED_NAMESPACE=wrapped_cub"
+    "THRUST_FORCE_32_BIT_OFFSET_TYPE")
```

Comment:
These header tests are linked to the interface targets built in `ThrustBuildCompilerTargets.cmake`, which already may have these options set, potentially to conflicting values.